loader/src/main/java/net/neoforged/fml/loading/ModDirTransformerDiscoverer.java

```diff
@@ -82,8 +82,11 @@ private static void scan(final Path gameDirectory) {
             // Skip if the mods dir doesn't exist yet.
             return;
         }
-        try (var walk = Files.walk(modsDir, 1)){
-            walk
+        try (var walk = Files.walk(modsDir, 1)) {
+            // Collect to list first, and then parallel stream it.
+            // Before JDK 19, Files.walk streams are not parallelized efficiently for small numbers of elements.
+            // See https://bugs.openjdk.org/browse/JDK-8280915.
+            walk.toList().stream()
```

Comment:
Why not `parallelStream` instead of `stream().parallel()`? They're likely equivalent but it feels like a code smell.

Reply:
Just because I didn't know it existed. :P There doesn't seem to be anything justifying it though in the code, but whatever.